substrate/client/src/client.rs

```diff
@@ -82,6 +94,32 @@ pub enum BlockStatus {
 	Unknown,
 }
 
+/// Block data origin.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum BlockOrigin {
+	/// Genesis block built into the client.
+	Genesis,
+	/// Block is part of the initial sync with the network.
+	NetworkInitialSync,
+	/// Block was broadcasted on the network.
+	NetworkBroadcast,
+	/// Block that was collated by this node.
+	Own,
```

Comment:
what does this mean in the context of the 2-phase commit?

Reply:
This means that the block body was generated on this machine and was not received from the network.

Reply:
the block body is almost always generated on another machine, but any block bodies coming directly from the consensus process have already been evaluated

Reply:
Added a variant for that.